app/javascript/Components/Result/image_viewer.jsx

```diff
@@ -1,34 +1,55 @@
 import React from "react";
-import {render} from "react-dom";
+import heic2any from "heic2any";
 
 export class ImageViewer extends React.PureComponent {
   constructor(props) {
     super(props);
     this.state = {
       rotation: window.start_image_rotation || 0,
       zoom: window.start_image_zoom || 1,
+      url: null,
     };
   }
 
   componentDidUpdate(prevProps, prevState) {
-    this.display_annotations();
-    this.adjustPictureSize();
-    this.rotateImage();
+    if (this.props.url && this.props.url !== prevProps.url) {
```

Comment:
Oh, sorry I realized that only the `setImageURL` should be guarded in this way. The `this.display_annotations`/`this.adjustPictureSize`/`this.rotateImage` should always trigger, otherwise the annotations won't show up properly. (Test in UI by creating an annotation and changing the zoom/rotation.)